### PR TITLE
fixed variable names that are named same as some std lib package names

### PR DESCRIPTION
**Pull request by varun06**
This is part of cleaning up some of the Lint and Vet issues. This is first PR in list of many PR's.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -28,3 +28,6 @@ profile.out
 
 simplest-uncommitted-msg-0.1-jar-with-dependencies.jar
 
+.idea
```
Comment (d1egoaz) [inline .gitignore:31]: This can be added in your global gitignore, this is very specific to your dev workflow

File: .gitignore [hunk 2]
```diff
@@ -28,3 +28,6 @@ profile.out
...
+
+
```
Comment (d1egoaz) [inline .gitignore:33]: these empty lines are not needed

File: async_producer_test.go [hunk 3]
```diff
@@ -1248,21 +1248,21 @@ func ExampleAsyncProducer_select() {
 	signals := make(chan os.Signal, 1)
 	signal.Notify(signals, os.Interrupt)
 
-	var enqueued, errors int
+	var enqueued, producerError int
```
Comment (d1egoaz) [inline async_producer_test.go:1251]: I think the plural version might read better `producerErrors`

File: async_producer_test.go [hunk 4]
```diff
@@ -1248,21 +1248,21 @@ func ExampleAsyncProducer_select() {
...
 			log.Println("Failed to produce message", err)
-			errors++
+			producerError++
 		case <-signals:
 			break ProducerLoop
 		}
 	}
 
-	log.Printf("Enqueued: %d; errors: %d\n", enqueued, errors)
+	log.Printf("Enqueued: %d; producerError: %d\n", enqueued, producerError)
```
Comment (d1egoaz) [inline async_producer_test.go:1265]: It doesn't seem necessary to replace errors here, 
this reads better:
```go
log.Printf("Enqueued: %d; errors: %d\n", enqueued, producerErrors)
```

File: async_producer_test.go [hunk 5]
```diff
@@ -1318,5 +1318,5 @@ ProducerLoop:
 
 	wg.Wait()
 
-	log.Printf("Successfully produced: %d; errors: %d\n", successes, errors)
+	log.Printf("Successfully produced: %d; producerError: %d\n", successes, producerError)
```
Comment (d1egoaz) [inline async_producer_test.go:1321]: same comment as above

File: consumer.go [hunk 6]
```diff
@@ -422,13 +422,13 @@ func (child *partitionConsumer) AsyncClose() {
 func (child *partitionConsumer) Close() error {
 	child.AsyncClose()
 
-	var errors ConsumerErrors
+	var consumerErrors ConsumerErrors
```
Comment (d1egoaz) [inline consumer.go:425]: yes, see how the plural version reads better here!

File: .gitignore (hunk 1 above)
Comment (varun06) [reply .gitignore:31]: yeah, I PR'd from a machine that doesn't have all the settings. will fix it now.

File: async_producer_test.go (hunk 4 above)
Comment (varun06) [reply async_producer_test.go:1265]: That's IDE refactoring, will fix it

**Pull request merged**